docs/source/extending.rst

```diff
@@ -0,0 +1,144 @@
+.. _extending:
+
+======================
+Extending the Notebook
```

Comment:
OK, that makes sense. In that case, let's make an extending folder, though, because this is going to be more information than we want in one document.